### PR TITLE
Add API instructions for /discount-codes

### PR DESCRIPTION
**Pull request by miguelcss**
### Notes
- Add API instructions for /discount-codes
- Endpoints added
```
/create
/list
/activate
/deactivate
/PUT /GET /DELETE event to discount
/PUT /GET /DELETE event + item to discount
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,5 +57,243 @@ External code not found - this code is not associated with any sold ticket in La
 ```
 
 
+## Discount codes
+Create new discount codes for specific events that you are the owner. Add and remove events and items to discount codes.
```
Comment (luiscrs14) [inline README.md]: I would replace the phrasing to `Create new discount codes for specific events that you are the owner **of**` or `Create new discount codes for specific events owned by you`

File: README.md [hunk 2]
```diff
@@ -57,5 +57,243 @@ External code not found - this code is not associated with any sold ticket in La
...
+discount_1 - event_50 - item_1
+discount_1 - event_50 - item_2
+```
+
+He then removes the 2 items with:
+```
+DELETE /discount-code/discount_1/event/event_50/item/item_1
+DELETE /discount-code/discount_1/event/event_50/item/item_2
+```
+What is the expected result of ?
```
Comment (luiscrs14) [inline README.md]: Is this intended to keep?

File: README.md [hunk 3]
```diff
@@ -57,5 +57,243 @@ External code not found - this code is not associated with any sold ticket in La
...
+### Activate
+Activate a discount code.
+```
+PUT /discount-code/<DISCOUNT_ID>/activate?api_key=<YOUR_API_KEY>
+```
+Response
+```
+"message": {
+    "discount_id": "DISCOUNT_ID",
+    "text": "Deactivate discount: DISCOUNT_ID",
```
Comment (luiscrs14) [inline README.md]: Is `text` here correct?

File: README.md [hunk 4]
```diff
@@ -57,5 +57,243 @@ External code not found - this code is not associated with any sold ticket in La
...
+### Deactivate
+Deactivate a discount code.
+```
+PUT /discount-code/<DISCOUNT_ID>/deactivate?api_key=<YOUR_API_KEY>
+```
+Reponse
+```
+"message": {
+    "discount_id": "DISCOUNT_ID",
+    "text": "Activated discount: DISCOUNT_ID",
```
Comment (luiscrs14) [inline README.md]: I think deactivate and activate are switched :)

File: README.md [hunk 5]
```diff
@@ -57,5 +57,243 @@ External code not found - this code is not associated with any sold ticket in La
...
+"message": {
+    "discount_id": "DISCOUNT_ID",
+    "event_id": "EVENT_ID",
+    "text": "Added event to discount",
+    "association_id": 12345
+},
+```
+
+### Get event
+Get the association id between an even and an existing discount code
```
Comment (luiscrs14) [inline README.md]: s/even/event

File: README.md [hunk 6]
```diff
@@ -57,5 +57,243 @@ External code not found - this code is not associated with any sold ticket in La
...
+```
+"message": {
+    "discount_id": "DISCOUNT_ID",
+    "event_id": "EVENT_ID",
+    "text": "Got event+item association to discount",
+    "associations": [
+        {
+            "id": 12345,
+            "discount_id": DISCOUNT_ID,
+            "events_id": EVENT_ID,
```
Comment (luiscrs14) [inline README.md]: Just noticed this now. Is this really `events_id`? Should it be `event_id`?

File: README.md [hunk 7]
```diff
@@ -57,5 +57,243 @@ External code not found - this code is not associated with any sold ticket in La
...
+            "item_id": 10
+        },
+        {
+            "id": 54321,
+            "discount_id": DISCOUNT_ID,
+            "events_id": EVENT_ID,
```
Comment (luiscrs14) [inline README.md]: Same

File: README.md (hunk 2 above)
Comment (miguelcss) [reply README.md]: changed a bit

File: README.md (hunk 4 above)
Comment (miguelcss) [reply README.md]: changed!

File: README.md (hunk 6 above)
Comment (miguelcss) [reply README.md]: yup!

File: README.md [hunk 8]
```diff
@@ -57,5 +57,245 @@ External code not found - this code is not associated with any sold ticket in La
...
+            "event_id": EVENT_ID,
+            "item_id": 20
+        }
+    ]
+},
+```
+
+
+### Delete event
+Delete the association id between an even and an existing discount code
```
Comment (luiscrs14) [inline README.md]: s/even/event

File: README.md [hunk 9]
```diff
@@ -57,5 +57,245 @@ External code not found - this code is not associated with any sold ticket in La
...
+```
+"message": {
+    "discount_id": "DISCOUNT_ID",
+    "event_id": "EVENT_ID",
+    "item_id": "ITEM_ID",
+    "text": "Got event+item association to discount",
+    "association": {
+        "id": EVENT_ID,
+        "discount_id": DISCOUNT_ID,
+        "events_id": EVENT_ID,
```
Comment (luiscrs14) [inline README.md:239]: More cases of `events_id`

File: README.md [hunk 10]
```diff
@@ -57,5 +57,245 @@ External code not found - this code is not associated with any sold ticket in La
...
+```
+Response sample
+```
+{
+    "stt": "OK",
+    "message": {
+        "event_id": "<EVENT_ID>",
+        "text": "Discount codes list.",
+        "discount_codes": "[
+            {\"discount_id\":12345,\"events_id\":500,\"item_id\":null,\"code\":\"SUPER\",\"type\":\"value\",\"value\":\"5.0000\",\"active\":\"yes\"},
```
Comment (luiscrs14) [inline README.md:106]: More cases of `events_id`

File: README.md (hunk 9 above)
Comment (miguelcss) [reply README.md:239]: This is correct, it comes from db query where the column is named `events_id`.

File: README.md (hunk 10 above)
Comment (miguelcss) [reply README.md:106]: same

**Review by luiscrs14 — APPROVED** [review]

**Review by luiscrs14 — APPROVED** [review]

**Pull request merged**